File: src/build_backend/post_requests/grade/grading_request.py
import requests
import json

# Define the URL for the Flask application's endpoint to get user answers
get_user_answers_api_url = "https://bitesize-grading-jsrdxhl2pa-ue.a.run.app/get-user-answers"
# Define the URL for the Flask application's endpoint to extract text
extract_text_api_url = "https://bitesize-grading-jsrdxhl2pa-ue.a.run.app/extract-text"
# Define the URL for the Flask application's endpoint to generate explanations
generate_explanations_api_url = "https://bitesize-grading-jsrdxhl2pa-ue.a.run.app/generate-explanations"

# Make a GET request to retrieve user answers
user_answers_response = requests.get(get_user_answers_api_url)

# Check if the request to get user answers was successful
if user_answers_response.status_code == 200:
    # Extract user answers data from the response
    user_answers = user_answers_response.json()
    
    # Make a GET request to the extract-text endpoint to get the parsed texts
    get_response = requests.get(extract_text_api_url)
    
    # Check if the GET request to extract text was successful
    if get_response.status_code == 200:
        # Extract the text data from the response
        parsed_texts = get_response.json().get('document_text', [])
        
        # Define the request payload using the extracted text and user answers
        payload = {
            "text": "\n".join(parsed_texts),  # Combine text from all files
            "user_answers": user_answers
        }
        
        # Send the POST request to the generate-explanations endpoint
        post_response = requests.post(generate_explanations_api_url, json=payload)
        
        # Check the response status code for the POST request
        if post_response.status_code == 200:
            # Print the JSON response with explanations
            response_data = post_response.json()
            print(json.dumps(response_data, indent=4))
        else:
            print(f"Error in POST request: {post_response.status_code} - {post_response.text}")
    else:
        print(f"Error in GET request to extract text: {get_response.status_code} - {get_response.text}")
else:
    print(f"Error in GET request to get user answers: {user_answers_response.status_code} - {user_answers_response.text}")

File: src/frontend/my-app/src/interactions/grading_request.js
export function getExtractedTextAndSendGradingRequest() {
  // Step 1: Make a GET request to the Flask app to retrieve the extracted text
  fetch('https://bitesize-question-gen-matz6gu77q-ue.a.run.app/extract-text')
    .then(response => {
      if (!response.ok) {
        throw new Error('Network response was not ok');
      }
      else{
        console.log('Initial response OK')
      }
      
      return response.json();
    })
    .then(data => {
      const extractedText = data.document_text.join("\n");
      const userAnswers = getUserAnswers();

      // Step 2: Construct the payload with the extracted text and user answers
      const payload = {
        text: extractedText,
        user_answers: userAnswers,
      };

      // Step 3: Make a POST request to the bitesize-grading service with the payload
      const gradingUrl = "https://bitesize-grading-matz6gu77q-ue.a.run.app/generate-explanations";
      return fetch(gradingUrl, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(payload),
      });
    })
    .then(response => {
      if (!response.ok) {
        throw new Error('Network response was not ok');
      }
      else{
        console.log('Generate Explantions OK')
      }
      return response.json();
    })
    .then(data => {
      console.log(data);
      // Handle the response data here
      
      // Step 4: Delete the contents of the bucket now that grading is done
      const bucketUrl = "https://bitesize-grading-matz6gu77q-ue.a.run.app/empty-bucket";
      return fetch(bucketUrl, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({
          bucket_name: "bitesize-documents-2",
          folder_name: "documents_to_be_summarized"
        }),
      });
    })
    .then(response => {
      if (!response.ok) {
        throw new Error('Failed to empty the bucket');
      }
      console.log('Bucket contents deleted successfully');
    })
    .catch(error => {
      console.error('Error:', error);
    });
}

function getUserAnswers() {
  fetch('https://bitesize-grading-matz6gu77q-ue.a.run.app/get-user-answers')
  .then(response => {
    if (!response.ok) {
      throw new Error('Network response was not ok');
    }
    return response.json();
  })
  .then(data => {
   return data;
  })

}
